ExSwift/Array.swift

```diff
@@ -1221,6 +1221,27 @@ internal extension Array {
     func sortUsing <U:Comparable> (block: ((T) -> U)) -> [T] {
         return self.sorted({ block($0.0) < block($0.1) })
     }
+    
+    /**
+        Sorts the array by the value returned from the block, in ascending order
+        Elements with nil values returned from the block appear at the end of the sorted array
+    
+        :param: the block to use to sort by
+        :returns: an array sorted by that block, in ascending order
+    */
+    
+    func sortUsing<U: Comparable>(block: (T) -> U?) -> [T] {
+        return sorted {
+            let value1 = block($0.0)
+            let value2 = block($0.1)
```

Comment:
In the current context $0.0 means essentially the same as $0. I left $0.0, $0.1 to be consistent with the existing sortUsing method's implementation.
$0 - means get the first parameter of closure. As a list of parameters is tuple, the first element of tuple is $0.0, and the second one $0.1.